src/pages/api/serviceList.ts

```diff
@@ -33,21 +33,28 @@ export default (req: NextApiRequestWithSession, res: NextApiResponse): void => {
             return;
         }
 
-        const selectedServices: string[] = [];
+        const selectedServices: SelectedService[] = [];
 
         const requestBody: { [key: string]: string | string[] } = req.body;
 
         Object.entries(requestBody).forEach(entry => {
-            const lineNameServiceCodeStartDate = entry[0];
+            const lineNameLineIdServiceCodeStartDate = entry[0];
             const description = entry[1];
             let serviceDescription: string;
             if (isArray(description)) {
                 [serviceDescription] = description;
             } else {
                 serviceDescription = description;
             }
-            const data = `${lineNameServiceCodeStartDate}#${serviceDescription}`;
-            selectedServices.push(data);
+            const data = `${lineNameLineIdServiceCodeStartDate}#${serviceDescription}`;
```

Comment:
This can also be simplified to:

```
const splitData = lineNameLineIdServiceCodeStartDate.split('#');
selectedServices.push({
    lineName: splitData[0],
    lineId: splitData[1],
    serviceCode: splitData[2],
    startDate: splitData[3],
    serviceDescription,
});
```